### ISSUE TITLE
[@warning "-34"] does not work

### ISSUE THREAD

**Issue by vicuna** [ocaml/ocaml#7886]
**Original bug ID:** 7886
**Reporter:** dinosaure
**Status:** acknowledged (set by @diml on 2019-01-07T13:22:55Z)
**Resolution:** open
**Priority:** normal
**Severity:** minor
**Category:** language features
**Monitored by:** @nojb @gasche

## Bug description

May be it's not an issue but I expect to discard the warning 34 in some pieces of my code instead to have a global `[@@@warning "-34"]` in my code. So, `type[@warning "-34"]` should work.

Currently on OCaml 4.07.0.

## Steps to reproduce

```
$ touch test.mli
$ echo 'type[@warning "-34"] foo = unit' > test.ml
$ ocamlc -w @34 test.mli test.ml
Error (warning 34): unused type foo.
```

**Comment by vicuna** [comment]
**Comment author:** @diml

A temporary workaround:

include struct
  [@@@warning "-34"]
  type foo = unit
end

**Comment by AndrewLitteken** [comment]
Is this still an available bug? I would be interested in working on this as an initial contribution.

**Comment by AndrewLitteken** [comment]
I believe that this bug is corrected in the current version of the repository. I cannot reproduce it

**Comment by Armael** [comment]
Indeed, looking through the Changes files, it looks like this was fixed in #1977. Thank you for pointing that out!
